sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/KeyVaultLoadStoreParameter.java

```diff
@@ -38,77 +32,86 @@ public final class KeyVaultLoadStoreParameter implements KeyStore.LoadStoreParam
+     * @param uri The Azure Key Vault URI.
+     * @param managedIdentity The managed identity.
      */
-    public KeyVaultLoadStoreParameter(String keyVaultUri, String managedIdentity) {
-        this(keyVaultUri, null, null, null, null, managedIdentity);
+    public KeyVaultLoadStoreParameter(String uri, String managedIdentity) {
+        this(uri, null, null, null, managedIdentity);
     }
 
     /**
      * Constructor.
      *
-     * @param keyVaultUri the Azure Key Vault URI.
-     * @param tenantId The tenant id.
-     * @param clientId The client id.
+     * @param uri The Azure Key Vault URI.
+     * @param tenantId The tenant ID.
+     * @param clientId The client ID.
      * @param clientSecret The client secret.
      */
-    public KeyVaultLoadStoreParameter(String keyVaultUri, String tenantId, String clientId, String clientSecret) {
-        this(keyVaultUri, null, tenantId, clientId, clientSecret, null);
+    public KeyVaultLoadStoreParameter(String uri, String tenantId, String clientId, String clientSecret) {
+        this(uri, tenantId, clientId, clientSecret, null);
     }
 
     /**
      * Constructor.
      *
-     * @param keyVaultUri the Azure Key Vault URI.
-     * @param tenantId The tenant id.
-     * @param clientId The client id.
+     * @param uri The Azure Key Vault URI.
+     * @param tenantId The tenant ID.
+     * @param clientId The client ID.
      * @param clientSecret The client secret.
-     * @param managedIdentity The Managed Identity.
+     * @param managedIdentity The managed identity.
      */
-    public KeyVaultLoadStoreParameter(String keyVaultUri, String tenantId, String clientId, String clientSecret, String managedIdentity) {
-        this(keyVaultUri, null, tenantId, clientId, clientSecret, managedIdentity);
+    public KeyVaultLoadStoreParameter(String uri, String tenantId, String clientId, String clientSecret,
+        String managedIdentity) {
+
+        this(uri, tenantId, clientId, clientSecret, managedIdentity, false);
     }
 
     /**
      * Constructor.
      *
-     * @param keyVaultUri the Azure Key Vault URI.
-     * @param loginUri The Azure login URI.
-     * @param tenantId The tenant id.
-     * @param clientId The client id.
+     * @param uri The Azure Key Vault URI.
+     * @param tenantId The tenant ID.
+     * @param clientId The client ID.
      * @param clientSecret The client secret.
-     * @param managedIdentity The Managed Identity.
+     * @param managedIdentity The managed identity.
+     * @param disableChallengeResourceVerification Indicates if challenge resource verification should be disabled.
      */
-    public KeyVaultLoadStoreParameter(String keyVaultUri, String loginUri, String tenantId, String clientId, String clientSecret, String managedIdentity) {
-        this.keyVaultUri = keyVaultUri;
-        this.loginUri = loginUri;
+    public KeyVaultLoadStoreParameter(String uri, String tenantId, String clientId, String clientSecret,
```

Comment:
It seems like this type has telescoping constructor pattern. Are these constructors used by end customers or just used internally within the library? If it's intended for customers, we should consider using setter methods for all these new optional params.

Reply:
My understanding is that these are never used by customers even if they are a part of the public package, as the KeyStore is normally initialized by adding the KV JCA security provider:
```java
KeyVaultJcaProvider provider = new KeyVaultJcaProvider();
Security.addProvider(provider);
```
and then calling
```java
KeyStore keyStore = KeyVaultKeyStore.getKeyVaultKeyStoreBySystemProperty();
```
or
```java
KeyStore keyStore = KeyStore.getInstance(KeyVaultJcaProvider.PROVIDER_NAME);
```
Do you think we should keep these public constructors in `KeyVaultKeyStore` untouched since they're technically part of the public API surface?

Reply:
I changed it to be a setter for public classes.